### PR TITLE
fix(theme-classic): autocollapse sidebar categories when navigating with paginator

### PR DESCRIPTION
**Pull request by pranabdas**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: Existing tests should cover the changes.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Auto collapse sidebar categories when navigating via paginator. Trying to fix https://github.com/facebook/docusaurus/issues/7387

### Test links

Deploy preview: https://deploy-preview-7411--docusaurus-2.netlify.app/

## Related issues/PRs
https://github.com/facebook/docusaurus/issues/7387

### CODE REVIEW

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Thanks for the fix! The paginator navigation issue seems resolved, but now, when you are in a category: https://deploy-preview-7411--docusaurus-2.netlify.app/docs/advanced/plugins/, the entire category is not collapsible, which seems like bad UX to me. Could you take a look?

**Comment by pranabdas** [comment]
@Josh-Cena Collapse button should be fine now, however one unintended change I see, if we click on non-category item it collapses previously opened category which used to be open prior to this change.

File: packages/docusaurus-theme-classic/src/theme/DocSidebarItem/Category.tsx [hunk 1]
```diff
@@ -38,12 +38,23 @@ function useAutoExpandActiveCategory({
...
-    if (justBecameActive && collapsed) {
+    if (autoCollapseCategories) {
+      if (justBecameActive && collapsed) {
+        setCollapsed(false);
+      } else if (!isActive && wasActive) {
+        setCollapsed(true);
+      }
+    } else if (justBecameActive && collapsed) {
       setCollapsed(false);
     }
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-classic/src/theme/DocSidebarItem/Category.tsx:56]:
```suggestion
    if (justBecameActive && collapsed) {
      setCollapsed(false);
    }
    if (autoCollapseCategories && !isActive && wasActive) {
      setCollapsed(true);
    }
```

**Comment by Josh-Cena** [comment]
> if we click on non-category item it collapses previously opened category

I noticed that as well. If you can manage to fix it it would be neat, otherwise I'll take a look later

**Comment by Josh-Cena** [comment]
Actually, it's because the fix is going in the wrong direction. We have two way to uncollapse a category: the "imperative" way through `setCollapsed`, and the "reactive" way through `updateCollapsed`, which also informs other categories to collapse themselves, you should pass the latter into `useAutoExpandActiveCategory` instead.

**Comment by pranabdas** [comment]
@Josh-Cena now the behaviour seems correct.

**Review by Josh-Cena — APPROVED** [review]
Thanks!

Easier than I thought 😄

**Pull request merged**